pkg/api/types.go

```diff
@@ -421,6 +425,19 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
```

Comment:
> On Apr 8, 2015, at 12:11 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > @@ -421,6 +425,19 @@ type NFSVolumeSource struct {
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> >  }
> >  
> > +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
> > +type GlusterfsVolumeSource struct {
> > +	// Required: EndpointsName is the endpoint name that details Glusterfs topology
> > +	EndpointsName string `json:"endpoints"`
> Perhaps the real question is whether you expect an external-to-kubernetes gluster cluster to be namespace-scoped or not?
> 
> A) The set of gluster endpoints is namespaced - use a headless service
> B) The set of gluster endpoints is not namespaced - use an object reference to a headless service
> C ?
> 
At least for persistent volumes (not namespaced), b) is required.

Only thing I can think of for C is "a DNS address" or "a list of ips".
> —
> Reply to this email directly or view it on GitHub.
>